### PR TITLE
Migration cpp20 improvements

### PR DESCRIPTION
**Pull request by Tanaka12**
* Changed headers so includes does not mess other client projects.
* Fixed conan package generator (utilities were being copied into the main package)

### CODE REVIEW

File: DbAdapterTestUtilities/Stubs/StubPrimaryKeyValue.cpp [hunk 1]
```diff
@@ -26,15 +26,15 @@ namespace systelab::db::test_utility {
 				switch(fieldType)
 				{
 					case db::BOOLEAN:
-						fieldValue.reset( new StubFieldValue(field.getName(), field.getBooleanDefaultValue()) );
+						fieldValue.reset( new StubFieldValue(field.getName(), std::make_optional(field.getBooleanDefaultValue())) );
```
Comment (Blackvals) [inline DbAdapterTestUtilities/Stubs/StubPrimaryKeyValue.cpp]: That's what I was afraid of when I saw that you removed the non-optional constructors.

File: DbAdapterTestUtilities/Stubs/StubRecord.h [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #pragma once
 
-#include "Mocks/MockRecord.h"
+#include "../Mocks/MockRecord.h"
```
Comment (Blackvals) [inline DbAdapterTestUtilities/Stubs/StubRecord.h]: I would rather go the other way around:
`#include "DbAdapterTestUtilities/Mocks/MockRecord.h`

As that much should be valid when exporting the conan package and is much more clear where things are.

File: ci/pipelines/Build.jenkinsfile [hunk 3]
```diff
@@ -51,8 +51,11 @@ pipeline
 					{
 						stage("Build ${configurations[i]}")
 						{
-							bat "conan export-pkg ${packageNameInterface}/conanfile.py ${packageNameInterface}/${version}@systelab/${channel} --profile ./package.profile --force"
-							dir("DbAdapterTestUtilities")
+							dir('DbAdapterInterface')
```
Comment (Blackvals) [inline ci/pipelines/Build.jenkinsfile]: Why? Are those changes aligned with gitignore and conanfile's export_sources patterns (of each project)?

File: DbAdapterTestUtilities/Stubs/StubFieldValue.h [hunk 4]
```diff
@@ -17,17 +17,12 @@ namespace systelab::db::test_utility {
...
-		StubFieldValue(const std::string& name, double value);
-		explicit StubFieldValue(const std::string& name, const std::string& value);
-		StubFieldValue(const std::string& name, const std::chrono::system_clock::time_point& value);
 
-		StubFieldValue(const std::string& name, const std::optional<int>& value);
-		StubFieldValue(const std::string& name, const std::optional<bool>& value);
-		StubFieldValue(const std::string& name, const std::optional<double>& value);
-		StubFieldValue(const std::string& name, const std::optional<std::string>& value);
-		StubFieldValue(const std::string& name, const std::optional<std::chrono::system_clock::time_point>& value);
+		explicit StubFieldValue(const std::string& name, const std::optional<int>& value);
```
Comment (Blackvals) [inline DbAdapterTestUtilities/Stubs/StubFieldValue.h]: The problem with this is that probably then you'll be force to especifically set an optional on the value side of things, which if the value was not meant to be optional is an abuse.
If you don't want to have so many constructors, maybe we can try either a template approach or the usage of std::variant (with proper order of types, like bool at the end) .

File: ci/pipelines/Build.jenkinsfile (hunk 3 above)
Comment (Tanaka12) [reply ci/pipelines/Build.jenkinsfile]: No, this was due to a problem on how the packet was being built. In this case, DbAdapterTestUtilities was being copied inside the packet as well. 

If I call conan inside the folder, only the specific headers of the project are copied into the packet

File: DbAdapterTestUtilities/Stubs/StubRecord.h (hunk 2 above)
Comment (Tanaka12) [reply DbAdapterTestUtilities/Stubs/StubRecord.h]: Solved

File: DbAdapterTestUtilities/Stubs/StubFieldValue.h (hunk 4 above)
Comment (Tanaka12) [reply DbAdapterTestUtilities/Stubs/StubFieldValue.h]: Makes sense, but due to time constraints I'll restore the initial constructors but moving the explicit qualifier to the bool constructor.

File: DbAdapterTestUtilities/Stubs/StubPrimaryKeyValue.cpp (hunk 1 above)
Comment (Tanaka12) [reply DbAdapterTestUtilities/Stubs/StubPrimaryKeyValue.cpp]: Yep, looks odd. As discussed offline, we are gonna revert this

File: ci/pipelines/Build.jenkinsfile (hunk 3 above)
Comment (Tanaka12) [reply ci/pipelines/Build.jenkinsfile]: Fixed by modifying how the conanfile is getting the headers

**Review by Blackvals — APPROVED** [review]

**Pull request merged**